esx_service/utils/auth_api.py

```diff
@@ -631,10 +637,20 @@ def _tenant_access_set(name, datastore, allow_create=None, volume_maxsize_in_MB=
         error_info = ErrorInfo(err_code, err_msg)
         return error_info
 
+    allow_create_val = str(allow_create)
+    allow_create_val = allow_create_val.lower()
```

Comment:
so the logic seems to be
- check if it's  a valid true/false string (case unsensitive)
- convert to bool
- pass to modify_privileges

in the code would be 
```python
  allow_create_bool, valid = convert_to_bool(allow_create)
 if not valid: 
   ...error msg... and return
 

 ...  pass allow_create_bool to modify privilege
```

right ? 

You code works ok, but reading it requires more effort :-)